### DISCUSSION TITLE
how to work with a flow that doesn't emit any value?

### DISCUSSION THREAD

**Discussion by AlexanderKott** [cashapp/turbine#170] [Q&A]
hello, thank you for this awesome library!

Can you tell me how to test flows that dont emit any values? 

i have 3 flows: 

val displayText = MutableSharedFlow<String?>()
val displayButtons = MutableSharedFlow<String?>()
val displayItemMessage = MutableSharedFlow<String?>()
...

My Test:

        val turbine1 = e.displayText.testIn(backgroundScope )
        val turbine2 = e.displayButtons.testIn(backgroundScope )
        val turbine3 = e.displayItemMessage.testIn(backgroundScope)

        game.displayNextStep()

        assertEquals("first value", turbine1.awaitItem())
        assertEquals("second value", turbine2.awaitItem())
        assert  that turbine3 doenst emit any values  (no values at all)     //HOW?

so i want to ensure turbine3 doen't have any values (nothing emmited) and continue my test. Now test just fails 
for turbine 3 : java.lang.AssertionError: No value produced in 1s if i write 
assertEquals(" -- ", turbine3.awaitItem() )

thank you!

**Comment by JakeWharton** [comment]
You can't really assert that a Flow will never emit an item. The best you can do is test that it has not _yet_ emitted an item (with `assertNoEvents()`. This, combined with the behavior of `runTest` which spins coroutines eagerly _should_ be enough to essentially test what you want.

**Reply by AlexanderKott** [reply]
Thank you very much for your answer. Could you please give me an axample of this : "The best you can do is test that it has not yet emitted an item (with assertNoEvents(). This, combined with the behavior of runTest which spins coroutines eagerly should be enough to essentially test what you want."

Turbine has expectNoEvents, but where to find assertNoEvents() ?

thank you

**Reply by JakeWharton** [reply]
Er, yes. That's the function I meant. It doesn't return anything, so it can become the third statement after the two calls to `awaitItem` to achieve what you want.

**Reply by AlexanderKott** [reply]
thank you, i will try
